### PR TITLE
Simplify ArchiveDB interface

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

- Simplifies the code
- Removes unnecessary exported functions
- Exports types to enable users to import the types correctly
- Simplifies key logic
- Ensures metadata prefix is conflict free

## How this was tested

CI

### CODE REVIEW

File: x/archivedb/batch.go [hunk 1]
```diff
@@ -3,52 +3,31 @@
...
+// batch is how a user performs modifications to the database.
 //
-// It takes any key and wraps it with dbKey. It is possible to construct many
-// batches but at Write() time, the height of the batch to be written to the
-// database must be the following height from the last written to the database,
-// otherwise an error will be thrown
+// It consumes puts and deletes at a specified height. When committing, an
+// atomic operation is created which registers the modifications at the
+// specified height and updates the last tracked height to be equal to this
+// batches height.
```
Comment (patrick-ogrady) [inline x/archivedb/batch.go:15]: nit: batch's

File: x/archivedb/key.go [hunk 2]
```diff
@@ -4,100 +4,79 @@
...
 //
-// All keys are prefixed with a VarUint64, which is the length of the keys, the
-// idea is to make reads even simpler, making it effectively a O(1) operation.
-func newDBKey(key []byte, height uint64) []byte {
+//	Example (Asumming heights are 1 byte):
+//	 |  User given  |  Stored as  |
+//	 |--------------|-------------|
+//	 |    foo:10    |  3:foo:245  |
+//	 |    foo:20    |  3:foo:235  |
+func newDBKey(key []byte, height uint64) ([]byte, []byte) {
```
Comment (aaronbuchwald) [inline x/archivedb/key.go:44]: could you add a comment for the return type?

File: x/archivedb/value.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package archivedb
+
+func newDBValue(value []byte) []byte {
+	dbValue := make([]byte, len(value)+1)
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply x/archivedb/value.go:7]: To make it more clear that the `heightKey` doesn't impact the key space. It isn't technically needed (I don't think)

File: x/archivedb/key.go [hunk 4]
```diff
@@ -4,100 +4,79 @@
...
+	return dbKey[:offset], dbKey[:prefixOffset]
 }
 
-// parseDBKey takes a slice of bytes and returns the inner key and the height
-func parseDBKey(rawKey []byte) ([]byte, uint64, error) {
-	rawKeyLen := len(rawKey)
-	if rawKeyLen < minInternalKeyLen {
-		return nil, 0, ErrInsufficientLength
+// parseDBKey takes a database formatted key and returns the user formatted key
+// along with its the height.
```
Comment (aaronbuchwald) [inline x/archivedb/key.go:58]:
```suggestion
// parseDBKey takes a database formatted key and returns the user formatted key
// along with its height.
```

File: x/archivedb/value.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply x/archivedb/value.go:7]: 🤔

[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply x/archivedb/value.go:7]: Ah, its to handle deletes somehow: 
image

[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply x/archivedb/value.go:7]: image

[the remark this answers is not in the document]
Comment (StephenButtolph) [reply x/archivedb/value.go:7]: ooooh sorry - I thought this was talking about the key space

[the remark this answers is not in the document]
Comment (StephenButtolph) [reply x/archivedb/value.go:7]: We differentiate between `Put(k, nil)` and `Delete(k)`

**Review by patrick-ogrady — APPROVED** [review]

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**